fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +190,413 @@ def dyncore_temporaries(shape):
-    # akap, ptop, pfull, n_map, comm):
-    grid = spec.grid
-    nonhydrostatic_pressure = utils.cached_stencil_class(
-        nh_p_grad.NonHydrostaticPressureGradient
-    )(cache_key="dyn_core_nhpgrad")
-    init_step = state.n_map == 1
-    end_step = state.n_map == spec.namelist.k_split
-    akap = constants.KAPPA
-    # peln1 = math.log(ptop)
-    # ptk = ptop**akap
-    dt = state.mdt / spec.namelist.n_split
-    dt2 = 0.5 * dt
-    hydrostatic = spec.namelist.hydrostatic
-    rgrav = 1.0 / constants.GRAV
-    n_split = spec.namelist.n_split
-    # TODO: Put defaults into code.
-    # m_split = 1. + abs(dt_atmos)/real(k_split*n_split*abs(p_split))
-    # n_split = nint( real(n0split)/real(k_split*abs(p_split)) * stretch_fac + 0.5 )
-    ms = max(1, spec.namelist.m_split / 2.0)
-    shape = state.delz.shape
-    # NOTE: In Fortran model the halo update starts happens in fv_dynamics, not here.
-    reqs = {}
-    if global_config.get_do_halo_exchange():
-        for halovar in [
-            "q_con_quantity",
-            "cappa_quantity",
-            "delp_quantity",
-            "pt_quantity",
-        ]:
-            reqs[halovar] = comm.start_halo_update(
-                state.__getattribute__(halovar), n_points=utils.halo
-            )
-        reqs_vector = comm.start_vector_halo_update(
-            state.u_quantity, state.v_quantity, n_points=utils.halo
-        )
-        reqs["q_con_quantity"].wait()
-        reqs["cappa_quantity"].wait()
-
-    state.__dict__.update(dyncore_temporaries(shape))
-    if init_step:
-        state.gz[:-1, :-1, :] = HUGE_R
-        state.diss_estd[grid.slice_dict(grid.compute_dict())] = 0.0
-        if not hydrostatic:
-            state.pk3[:-1, :-1, :] = HUGE_R
-    state.mfxd[grid.slice_dict(grid.x3d_compute_dict())] = 0.0
-    state.mfyd[grid.slice_dict(grid.y3d_compute_dict())] = 0.0
-    state.cxd[grid.slice_dict(grid.x3d_compute_domain_y_dict())] = 0.0
-    state.cyd[grid.slice_dict(grid.y3d_compute_domain_x_dict())] = 0.0
-
-    if not hydrostatic:
-        # k1k = akap / (1.0 - akap)
+class AcousticDynamics:
+    """
+    Fortran name is dyn_core
+    Peforms the Lagrangian acoustic dynamics described by Lin 2004
+    """
 
-        # To write in parallel region, these need to be 3D first
-        state.dp_ref = utils.make_storage_from_shape(
-            shape, grid.full_origin(), cache_key="dyn_core_dp_ref"
+    def __init__(self, comm: fv3gfs.util.CubedSphereCommunicator, namelist):
+        self.comm = comm
+        self.namelist = namelist
+        assert self.namelist.d_ext == 0
+        assert self.namelist.beta == 0
+        assert not self.namelist.use_logp
+        self.grid = spec.grid
+        self.do_halo_exchange = global_config.get_do_halo_exchange()
+        self._nk_heat_dissipation = get_nk_heat_dissipation(namelist, self.grid)
+        self.nonhydrostatic_pressure = nh_p_grad.NonHydrostaticPressureGradient()
```

Comment:
Can you rename this to something like `nonhydrostatic_pressure_gradient`, `nh_pressure_gradient`, or `nonhydrostatic_pg`? The "gradient" part is an important distinction, "nonhydrostatic_pressure" could be misleading. If this name is too long, it is probably better to shorten parts of it, and at least have it documented in the class name/attribute type. "nonhydrostatic pressure" to me is `p'` where pressure is defined as `p = p_hs + p'` and `p_hs` is the hydrostatic component of pressure.

Not necessarily required right now, but also nh_p_grad applies the pressure gradient force/PGF, as opposed to computing a pressure gradient, so a full name would probably be something like `NonHydrostaticPressureGradientForce`. Maybe `PressureGradientForce` is enough for us, since we are explicitly writing a nonhydrostatic model? `self.pressure_gradient_force`? It makes it clearer that this is the full hydrostatic + nonhydrostatic component, whereas "nonhydrostatic pressure gradient force" kind of says you're leaving out the hydrostatic part.

Reply:
Interesting follow-up question: is `p_grad_c` nonhydrostatic?

Reply:
sure